.github/workflows/build.yml

```diff
@@ -0,0 +1,63 @@
+name: CI
+
+on:
+  # Only run push on master
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - '*.md'
+  # Always run on PRs
+  pull_request:
+    paths-ignore:
+      - '*.md'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        java-version:
+          - 1.7
+          - 1.8
+          - 9
+          - 10
+          - 11
+          - 12
+          - 13
+          - 14
+          - 15-ea
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Validate Gradle Wrapper
+        uses: gradle/wrapper-validation-action@v1
+
+      - name: Generate cache key
+        run: ./.github/workflows/checksum.sh checksum.txt
```

Comment:
Skeptical of this complexity compared to its value.

Reply:
we could manually implement it, the main issue is that their examples only cover .gradle files and not properties or wrapper changes

Reply:
I mean caching things probably isn't worth it. We either download 100MB from the cache or we download 100MB from Maven central and Gradle which is also just a cache. What's the difference except one requires no additional build configuration to maintain.

Reply:
The github cache is significantly faster in my experience, compared to waiting for gradle to figure it out and roundtrip to central. I don't have strong feelings on it, but think it's marginally worth it for the relatively low cost

Reply:
A good comparison from the runs on my repo that I linked below: cached build took ~2:20, cache misses took ~4min